src/watchdog/src/watchdog.py

```diff
@@ -340,9 +346,15 @@ def load_machine_list(configFilePath):
         return yaml.load(f)["hosts"]
 
 
-def request_with_histogram(url, histogram):
+def request_with_histogram(url, histogram, tls, ca_path, bearer_path):
     with histogram.time():
-        return requests.get(url).json()
+        if tls:
+            with open(bearer_path, 'r') as bearer_file:
```

Comment:
Same above, read the file in main and pass content to this function. Since two function needs the same content, I think maybe you should pass headers directly, this save the cost of creating map over and over again.

Reply:
pass headers